plugins/out_s3/s3.c

```diff
@@ -841,6 +873,39 @@ static int cb_s3_init(struct flb_output_instance *ins,
         }
     }
 
+    tmp = flb_output_get_property("server-side-encryption-aws-kms-key-id", ins);
+    if (tmp) {
+        ctx->server_side_encryption_aws_kms_key_id = (char *) tmp;
+    }
+
+    tmp = flb_output_get_property("server-side-encryption", ins);
+    if (tmp) {
+        ctx->server_side_encryption = (char *) tmp;
+    }
+
+    /* Validate sse values */
+    if (ctx->server_side_encryption != NULL) {
+        if (strcmp(ctx->server_side_encryption, "AES256") != 0 &&
```

Comment:
We needn't to distinguish upper case or lower case.
So, we need to use `strcasecmp` instead of `strcmp`.

Plus, we need to verify with length, too.
On paper, it needs to use `strncasecmp` here.